### PR TITLE
test on current python/django versions, adopt ruff

### PR DESCRIPTION
**Pull request by chris48s**
This PR is quite similar in nature to https://github.com/DemocracyClub/dc_django_utils/pull/90 so I am going to annotate it a bit more lightly.

Focus for this one is:

- Run the tests on Python 3.10/3.12, drop any previous versions
- Run the tests on Django 4.2/5.1, drop any previous versions
- Ditch black, adopt ruff
- Run tests against Postgres 16

For this one I am leaving the package manifest as setup.py. Also, we don't need to worry about djhtml here because there are no templates.

Because:

- We were running black with the default line length (88) but the 'standard' DC ruff config sets the line length to 80 so this meant that ruff format wanted to make lots of whitespace changes
- We never set up isort or flake8 on this project, so there were a bunch of new errors for ruff to flag

There are quite a lot of code formatting changes in this PR

- 8e7079d contains all the whitespace changes and safe auto-fixes, so probably doesn't need much review
- a80db47 are all the changes where I manually changed code to make ruff happy. Sorry - I ideally should have split that into 5 smaller commits to make it easier to review.

### CODE REVIEW

File: uk_geo_utils/helpers.py
Comment (chris48s) [inline uk_geo_utils/helpers.py:246]: ruff auto-generated the line

`if end_suffix and end_number or start_number and end_number:`

in 8e7079d by collapsing this redundant `elif`

Ruff is correct that `and` has a higher order of precedence than `or`, but I find this a lot easier to parse quickly with brackets.

**Comment by chris48s** [comment]
TODO: once this is approved, cut a new release after merging

**Comment by VirginiaDooley** [comment]
I installed this locally into EE, all fine except same questions as mentioned in the similar [dc_utils PR](https://github.com/DemocracyClub/dc_django_utils/pull/90#issuecomment-2324426912).

**Review by VirginiaDooley — APPROVED** [review]

**Pull request merged**
